### DISCUSSION TITLE
How can we set up a caching simulation experiment with a fixed cache size in GBs/MBs?

### DISCUSSION THREAD

**Discussion by super-shayan** [ben-manes/caffeine#1415] [Q&A] [answered]
It seems that in the configuration file, we can set the total number of items to cache [here](https://github.com/ben-manes/caffeine/blob/808d3d5ea2408b795b1fffdfbef5240489efc57f/simulator/src/main/resources/reference.conf#L34), but I wanted to ask if it was possible to do that with regard to cache size in GBs/MBs etc.

**Comment by ben-manes** [comment, marked as answer]
Some of the caches and traces support a weighted size, were you specify that value in terms of bytes. Those policies and traces are annotated with that characteristic, e.g. [CaffeinePolicy](https://github.com/ben-manes/caffeine/blob/808d3d5ea2408b795b1fffdfbef5240489efc57f/simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/policy/product/CaffeinePolicy.java#L38) and [TwitterTraceReader](https://github.com/ben-manes/caffeine/blob/808d3d5ea2408b795b1fffdfbef5240489efc57f/simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/twitter/TwitterTraceReader.java#L41-L43). However, I have not spent a lot of time on weighted size evaluations so there may be improvements needed, e.g. the simulator's TinyLFU sketch naively [presizes](https://github.com/ben-manes/caffeine/blob/808d3d5ea2408b795b1fffdfbef5240489efc57f/simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/admission/countmin4/CountMin4.java#L54) assuming a fixed entry count whereas Caffeine will grow it as needed. A lot of these fixes were made in a @ohadeytan's [size research fork](https://github.com/ohadeytan/caffeine/tree/size_research), but we never did the work to upstream it. It would be good to do someday, as well as incorporate his ideas into the library.

**Reply by super-shayan** [reply]
perfect! thanks alot!! =)
